Issues #552, #557, #558

mlantz opened the pull request:
Just a handful of bug fixes.

gtanner commented:
When the user cancels without selecting a picture we need to make sure we handle it correctly on each platform:

http://docs.phonegap.com/en/2.1.0/cordova_media_capture_capture.md.html#capture.captureImage
If the operation is terminated by the user before an image is captured, the CaptureErrorCB callback will be invoked with a CaptureError object with the CaptureError.CAPTURE_NO_MEDIA_FILES error code.

https://developer.blackberry.com/html5/apis/blackberry.media.camera.html#.takePicture
should invoke on camera closed event

https://developer.blackberry.com/html5/apis/blackberry.invoke.card.html#.invokeCamera
Should call the onCancel callback

mlantz commented:
Thanks good call.

Commit:
Issue #556

One possible solution to this problem is to completely remove it, and
put it back.

mlantz commented:
Show me how. I'd love to figure that stuff out.

Commit:
Issue #558

Commit:
Issue #557

Added the proper callback error code for Cordova, when an image is not
selected.

Commit:
Issue #552

mlantz commented:
After a bundle of learning the magic of rebase. We have some actual solutions here!

brentlintner commented:
@mlantz I noticed there were still some lint and upstream issues with some of the commits (ex: error argument was removed from function sig in 557 fix, which did not put it back). I've fixed them all, and have them [here](https://github.com/brentlintner/ripple-ui/commits/bug-fixes/) for reference (I also temporarily reworded them so they did not show up in respective issues)- just need to make sure they are in working order (plan is to do that tomorrow). If you want to fix the originals up and push, that is cool. Either way, once testing is double checked, I will get these in ASAP! :-)

gtanner commented on lib/ripple/ui/plugins/camera.js:
```diff
@@ -20,6 +20,7 @@ var ui = require('ripple/ui'),
     upload = document.getElementById('picture-upload'),
     select = document.getElementById('select-file'),
     take = document.getElementById('take-file'),
+    optOut = document.getElementById('camera-cancel'),
```
I don't think you included your changes to overlay.html to add the cancel button
